packages/dashboard-frontend/src/services/workspaceAdapter/index.ts

```diff
@@ -200,7 +200,14 @@ export class WorkspaceAdapter<T extends che.Workspace | IDevWorkspace> implement
   set storageType(type: che.WorkspaceStorageType) {
     if (isWorkspaceV1(this.workspace)) {
       const attributes = typeToAttributes(type);
-      Object.assign(this.workspace.devfile.attributes, attributes);
+      if (!this.workspace.devfile.attributes) {
+        this.workspace.devfile.attributes = {};
+      }
+      if (attributes) {
+        Object.assign(this.workspace.devfile.attributes, attributes);
```

Comment:
The result of this 'Object.assign(this.workspace.devfile.attributes, attributes)' could bring a wrong storage type'.
It is better to remove 'persistVolumes' and 'asyncPersist' keys, then set needed.

Reply:
That makes sense, I'll provide a fix.

Reply:
done